synapse/api/auth.py

```diff
@@ -78,6 +78,18 @@ def __init__(self, hs):
         register_cache("cache", "token_cache", self.token_cache)
 
         self._account_validity = hs.config.account_validity
+        self._track_appservice_user_ips = self.hs.config.track_appservice_user_ips
+        self._macaroon_secret_key = self.hs.config.macaroon_secret_key
+        self._server_notices_mxid = self.hs.config.server_notices_mxid
+        self._hs_disabled = self.hs.config.hs_disabled
+        self._limit_usage_by_mau = self.hs.config.limit_usage_by_mau
+        self._hs_disabled_message = self.hs.config.hs_disabled_message
+        self._admin_contact = self.hs.config.admin_contact
+        self._limit_usage_by_mau = self.hs.config.limit_usage_by_mau
+        self._mau_limits_reserved_threepids = (
+            self.hs.config.mau_limits_reserved_threepids
+        )
+        self._max_mau_value = self.hs.config.max_mau_value
```

Comment:
can we factor out a separate class rather than add all these fields to this big class?

Reply:
How do you mean? Moving the methods that use these to a new class or putting all these config options in a new inline class?

Reply:
move the methods to a new class. dunno how easy that is

Reply:
It's not too bad to move, I just need to update all the references to it.

I could start with just having `Auth.check_auth_blocking` call the new method though.

Or I could just go ahead and update all the references. My plan is:

* Create a new class `AuthBlocking`
* It just handles this method
* Have to add a `build_auth_blocking` method to the hs in addition to `build_auth`.

Before I get too far into any of this, does this sound sane or should we back out?

Reply:
> I could start with just having `Auth.check_auth_blocking` call the new method though.

sounds sensible for now

>     * Create a new class `AuthBlocking`
>     * It just handles this method

yes

>     * Have to add a `build_auth_blocking` method to the hs in addition to `build_auth`.

I'd have `Auth.__init__` build the new object

Reply:
Unfortunately there's quite a few tests that changed the hs's config after the hs had already been created, which need fixing up.